src/main/java/com/google/devtools/build/lib/bazel/bzlmod/BzlmodRepoRuleCreator.java

```diff
@@ -93,4 +99,52 @@ public static Rule createRule(
     }
     return rule;
   }
+
+  public static void validateLabelAttrs(Rule rule, ModuleExtensionId owningExtension)
+      throws EvalException {
+    for (var attribute : rule.getRuleClassObject().getAttributes()) {
+      Object value = rule.getAttr(attribute.getName());
+      Collection<?> toValidate =
+          switch (value) {
+            case List<?> list -> list;
+            case Map<?, ?> map -> map.keySet();
+            case null, default -> singletonList(value);
+          };
+      for (var item : toValidate) {
+        if (!(item instanceof Label label)) {
+          continue;
+        }
+        if (label.getRepository().isVisible()) {
+          continue;
+        }
+        String unprefixedRepoName =
+            label.getRepository().getName().substring(label.getName().lastIndexOf('~') + 1);
+        RepositoryName owningModuleRepoName = owningExtension.getBzlFileLabel().getRepository();
+        String owningModule;
```

Comment:
this part assumes that `label.getRepository().getOwnerRepo()` is the same as `owningModuleRepoName`, which is not always true (e.g. the 'bad label passed to tag and forwarded to repo attr' case). could we maybe only generate the following errors when the two do match?

Reply:
I modified the message and also check tag attributes now.